### PR TITLE
Customizable scorer

### PR DESCRIPTION
**Pull request by nicolas-chaulet**
Improving pointgroup, but still problems with an exploding loss when the scorer gets activated

### CODE REVIEW

**Comment by nicolas-chaulet** [comment]
@humanpose1 @tchaton this fixes the bug with scoring, PointGroup is now operational!

File: test/test_pointgroup.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import os
+import sys
+import unittest
+import torch
+
+ROOT = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..")
+sys.path.insert(0, ROOT)
+
+from torch_points3d.models.panoptic.pointgroup import offset_loss, instance_iou_loss
```
Comment (tchaton) [inline test/test_pointgroup.py:9]: Could you move those losses to panoptic loss. They are not related to pointgroup specifically.

File: torch_points3d/models/panoptic/pointgroup.py [hunk 2]
```diff
@@ -1,16 +1,60 @@
...
+
+def instance_iou_loss(
+    predicted_clusters: List[torch.Tensor],
+    cluster_scores: torch.Tensor,
+    instance_labels: torch.Tensor,
+    batch: torch.Tensor,
+    min_iou_threshold=0.25,
+    max_iou_threshold=0.75,
+):
+    """ Loss that promotes higher scores for clusters wuth higher instance iou,
```
Comment (tchaton) [inline torch_points3d/models/panoptic/pointgroup.py:42]: typo with

File: torch_points3d/models/panoptic/pointgroup.py [hunk 3]
```diff
@@ -77,20 +128,37 @@ def forward(self, epoch=-1, **kwargs):
...
-                for i, cluster in enumerate(all_clusters):
-                    x.append(backbone_features[cluster])
-                    coords.append(self.input.coords[cluster])
-                    batch.append(i * torch.ones(cluster.shape[0]))
-                batch_cluster = Data(x=torch.cat(x).cpu(), coords=torch.cat(coords).cpu(), batch=torch.cat(batch).cpu())
-                score_backbone_out = self.Scorer(batch_cluster)
-                if self._scorer_is_encoder:
-                    cluster_feats = score_backbone_out.x
+                if self._activate_scorer:
+                    x = []
```
Comment (tchaton) [inline torch_points3d/models/panoptic/pointgroup.py:132]: Can you wrap this within its own function ? Long forward are pretty hard to follow.

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,7 +5,7 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [Unreleased]
+## 1.1.0
```
Comment (tchaton) [inline CHANGELOG.md:8]: Add PointGroup to readme, please !
